### PR TITLE
Merge changes from onprc19.1Prod from 66140-66173

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
ONPRC is about to upgrade to 20.11, so we want to be sure they have everything from their 19.1 SVN branch.

#### Changes
* See SVN history

### CODE REVIEW

File: onprc_ehr/resources/views/groupProcessing.html [hunk 1]
```diff
@@ -241,6 +319,16 @@
...
+
+                var field = btn.up('panel').down('#animalField');
+                var animalid = field.getValue() || [];
+                if (!Ext4.isEmpty(animalid)){
+                animalid = Ext4.String.trim(animalid);
+                animalid = animalid.replace(/[\s,;]+/g, ';');
+                animalid = animalid.replace(/(^;|;$)/g, '');
+                }
+                animalid = Ext4.isArray(animalid) ? animalid: [animalid];
+
```
Comment (labkey-bpatel) [inline onprc_ehr/resources/views/groupProcessing.html:331]: lines 323-331 duplicated a few times above (lines 94-101, 141-148, etc.) - maybe extract it into a new method.

File: onprc_ehr/resources/web/onprc_ehr/window/BulkSerologyScanWindow.js [hunk 2]
```diff
@@ -169,11 +169,12 @@ Ext4.define('ONPRC_EHR.window.BulkSerologyScanWindow', {
...
         }
-        //var project = this.resolveProjectByName(Ext4.String.trim(row[2]), errors, rowIdx);
 
 
-        var project = this.resolveProjectByName(Ext4.String.trim(row[13]), errors, rowIdx);
-        var name = Ext4.String.trim(row[14]);
+        var project = this.resolveProjectByName(Ext4.String.trim(row[14]), errors, rowIdx);
+
+        var name = Ext4.String.trim(row[15]);
+
```
Comment (labkey-bpatel) [inline onprc_ehr/resources/web/onprc_ehr/window/BulkSerologyScanWindow.js:177]: hard coded indexes on line 174 (row[14]) & line 176(row[15]) -- Maybe a question for onprc folks if this is working as expected.

File: onprc_ehr/resources/web/onprc_ehr/window/BulkSerologyScanWindow.js [hunk 3]
```diff
@@ -221,12 +223,16 @@ Ext4.define('ONPRC_EHR.window.BulkSerologyScanWindow', {
...
                     agent: Ext4.String.trim(parsed[5][k].substr(parsed[5][k].length - 7, 7)),
                     method: method,
                     result: Ext4.String.trim(row[k]),
-                    qualifier: Ext4.String.trim(row[15]),
-                    remark: Ext4.String.trim(row[12]),
+
+                    qualifier: Ext4.String.trim(row[16]),
+
+                    remark: Ext4.String.trim(row[13]),
+
```
Comment (labkey-bpatel) [inline onprc_ehr/resources/web/onprc_ehr/window/BulkSerologyScanWindow.js:235]: same comment as above about hardcoded index values.

File: onprc_ehr/resources/queries/study/BCSScoreWeights.sql [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+FROM study.Clinical_Observations p, study.weight K
+WHERE p.qcstate.publicdata = true
+  And (p.id = k.id)
+  And p.category = 'BCS'
+  And p.date in (Select max(r.date) from study.Clinical_Observations r, study.Weight j
+                 Where r.category = 'BCS' And r.id = p.id And r.id = j.id
+                   And (TIMESTAMPDIFF('SQL_TSI_DAY', j.date, r.date) < 8) And r.qcstate = 18 And j.qcstate = 18)
+
+  And (K.date in (select min(s.date) as date  from study.Weight s Where s.Id = k.Id And ( (TIMESTAMPDIFF('SQL_TSI_DAY', s.date, p.date) <= 0) And (TIMESTAMPDIFF('SQL_TSI_DAY', s.date, p.date) > -8) ) And s.qcstate = 18)
+    or K.date in (select max(s.date) as date  from study.Weight s Where s.Id = k.Id And ( (TIMESTAMPDIFF('SQL_TSI_DAY', s.date, p.date) < 8) And (TIMESTAMPDIFF('SQL_TSI_DAY', s.date, p.date) > 0) ) And s.qcstate = 18)  )
```
Comment (labkey-bpatel) [inline onprc_ehr/resources/queries/study/BCSScoreWeights.sql:37]: hardcoded value for qcstate (s.qcstate = 18 on line 34, 36, 37) -- is it possible to compare the label value instead (ex. s.qcstate.Label = 'completed')

**Review by labkey-bpatel — APPROVED** [review]
Approved. Just a few comments though if onprc devs would to make those minor changes.

**Comment by jonesgaohsu** [comment]
Team at ONPRC has validated the code with No Issues

**Pull request merged**